source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/tasks/index.tsx

```diff
@@ -1,15 +1,138 @@
+import './tasks.scss';
+import { uuid } from '@orbit/utils';
+
+const prettyMS = require('pretty-ms');
+
+interface Task {
+  type: string,
+  id: string,
+  attributes: TaskAttributes
+}
+
+export interface IOwnProps {
+  tasks: Task[]
+}
+export type IProps =
+  & IOwnProps
+  & WithDataProps
+
+class Tasks extends React.Component<IProps> {
 
   state = { data: {}, errors: {} };
 
-  render() {
+  //TODO: Remove this method when we collect tasks from the backend
+  generateRandomTaks = async () => {
+
+    const products = ['Android APK w/Embedded Audio','HTML website'];
+    const status = ['Awaiting Approval','Send / Recieve', 'App Store Preview'];
+
+    await this.props.updateStore(t => t.addRecord({
+      type: TYPE_NAME,
+      id: uuid(),
+      attributes: {
+        project: 'Example Bible',
+        product: products[Math.floor(Math.random() * (2))],
+        status: status[Math.floor(Math.random() * (3))],
+        waitTime: Math.floor(Math.random() * (1296001))
+      }
+    }));
+  }
+
+  componentWillMount() {
+    this.generateRandomTaks();
+  }
+
+  productIcon = (productName) => {
+
+    const icons = [{id: 'android', icon:'android'},{id:'html', icon: 'file code'}];
+
+    for (let i=0; i < icons.length; i++) {
+      if (productName.toLowerCase().includes(icons[0].id)) {
+        return <Icon className={icons[0].icon}/>
+      }
+    }
+
     return null;
   }
+
+  render() {
+
+    const { tasks } = this.props;
+
+    console.log(tasks);
+
+    return (
+      <Container className='tasks'>
+        <h1 className='page-heading'>My Tasks</h1>
+        {
+          tasks ?
+            <Table>
+              <Table.Header>
+                <Table.Row>
+                  <Table.HeaderCell>Project</Table.HeaderCell>
+                  <Table.HeaderCell>Product</Table.HeaderCell>
+                  <Table.HeaderCell>Assigned To</Table.HeaderCell>
+                  <Table.HeaderCell>Status</Table.HeaderCell>
+                  <Table.HeaderCell>Wait Time</Table.HeaderCell>
+                  <Table.HeaderCell></Table.HeaderCell>
+                </Table.Row>
+              </Table.Header>
+
+              <Table.Body>
+                {tasks.map((task,index) => {
+
+                  const { project, product, status, waitTime, user } = task.attributes;
+
+                  return (
+                    <Table.Row key={index}>
+                      <Table.Cell>
+                        <Link to={`projects/${task.id}`}>{project}</Link>
+                      </Table.Cell>
+                      <Table.Cell>
+                        { this.productIcon(product) }
+                        <span>{product}</span>
+                      </Table.Cell>
+                      <Table.Cell>{user ? `${user.name}` : '[unclaimed]'}</Table.Cell>
+                      <Table.Cell className='red'>{status}</Table.Cell>
+                      <Table.Cell>
+                        <span>{prettyMS(waitTime, { secDecimalDigits: 0 })}</span>
```

Comment:
gotchya, prettyMS is for time stuff.
coolio

Reply:
yup